### DISCUSSION TITLE
how to print traceid when only 10% sampling

### DISCUSSION THREAD

**Discussion by DongWeiMa** [apache/skywalking#9811] [General]
When sampling is only 10%, the traceId of 90% of the logs is "Ignored_Trace",
But what we expect is that even if it is not collected, there is an independent traceid, so that we can better troubleshoot the problem.
Otherwise, in the case of Ignored_Trace, it is difficult to troubleshoot the problem

**Comment by heyanlong** [comment]
All issues and discussions on GitHub must be in English.

**Reply by DongWeiMa** [reply]
👌

**Reply by DongWeiMa** [reply]
When sampling is only 10%, the traceId of 90% of the logs is "Ignored_Trace",
But what we expect is that even if it is not collected, there is an independent traceid, so that we can better troubleshoot the problem.
Otherwise, in the case of Ignored_Trace, it is difficult to troubleshoot the problem

**Reply by DongWeiMa** [reply]
so wo can i do? only not set sampling?

**Comment by DongWeiMa** [comment]
modify IgnoredTracerContext.java



    private final String traceId;
    public IgnoredTracerContext() {
        this.stackDepth = 0;
        this.correlationContext = new CorrelationContext();
        this.extensionContext = new ExtensionContext();
        this.traceId = GlobalIdGenerator.generate();;
    }

    @Override
    public String getReadablePrimaryTraceId() {
        return traceId;
    }

`

**Comment by wu-sheng** [comment]
We only recommend sampling on the backend if you want all data. Because even you tried like you mentioned, all traffic and metrics relative to those ignored traces are gone as well.

**Reply by DongWeiMa** [reply]
For some business problems, many times the problem can be solved only by logs.
There is also the issue of cost. For 100% monitoring, the cost will increase several times, and it will bring performance overhead.

**Reply by DongWeiMa** [reply]
It is a better choice to dynamically turn on full collection in emergency situations

**Reply by wu-sheng** [reply]
I think you don't follow my point. I just recommend sampling on the backend, not asking for 100% sampling.

**Reply by DongWeiMa** [reply]
sorry,so  what should I do?

**Reply by wu-sheng** [reply]
Follow this OAP doc, https://skywalking.apache.org/docs/main/next/en/setup/backend/trace-sampling/#trace-sampling-at-server-side

I think 100% sampling is not the issue at the agent side, because your change basically are doing the same change.

**Reply by DongWeiMa** [reply]
I didn't understand correctly just now that "sampling on the backend " means “at server side ” and don't know can  set Trace Sampling at server side ,so sorry.

We are using Alibaba Cloud's Tracing Analysis and i found the place to configure. 
Thinks.
